cosmic/data_source_cosmic_network_acl.go

```diff
@@ -0,0 +1,112 @@
+package cosmic
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/MissionCriticalCloud/go-cosmic/v6/cosmic"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceCosmicNetworkACL() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCosmicNetworkACLRead,
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+
+			// Computed values
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCosmicNetworkACLRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*cosmic.CosmicClient)
+
+	p := conn.NetworkACL.NewListNetworkACLListsParams()
+	p.SetListall(true)
+
+	cosmicACLLists, err := conn.NetworkACL.ListNetworkACLLists(p)
+	if err != nil {
+		return fmt.Errorf("Failed to list ACL Lists: %s", err)
+	}
+
+	filters := d.Get("filter")
+	var ACLLists []*cosmic.NetworkACLList
+
+	for _, l := range cosmicACLLists.NetworkACLLists {
+		match, err := applyACLRuleFilters(l, filters.(*schema.Set))
+		if err != nil {
+			return err
+		}
+
+		if match {
+			ACLLists = append(ACLLists, l)
+		}
+	}
+
+	if len(ACLLists) == 0 {
+		return fmt.Errorf("No Network ACL List matched with the specified filter")
+	}
+
+	if len(ACLLists) > 1 {
+		return fmt.Errorf("More than one Network ACL List found, use a more specific filter")
+	}
+
+	ACLList := ACLLists[0]
+	log.Printf("[DEBUG] Selected Network ACL List: %s\n", ACLList.Id)
+
+	return networkACLAttributes(d, ACLList)
+}
+
+func networkACLAttributes(d *schema.ResourceData, list *cosmic.NetworkACLList) error {
+	d.SetId(list.Id)
+	d.Set("name", list.Name)
+	d.Set("description", list.Description)
+	d.Set("vpc_id", list.Vpcid)
+
+	return nil
+}
+
+func applyACLRuleFilters(list *cosmic.NetworkACLList, filters *schema.Set) (bool, error) {
+	var ACLListJSON map[string]interface{}
+	t, _ := json.Marshal(list)
+	json.Unmarshal(t, &ACLListJSON)
+
+	for _, f := range filters.List() {
+		m := f.(map[string]interface{})
+
+		r, err := regexp.Compile(m["value"].(string))
+		if err != nil {
+			return false, fmt.Errorf("Invalid regex: %s", err)
+		}
+
+		ACLListField := ACLListJSON[m["name"].(string)].(string)
+		if !r.MatchString(ACLListField) {
+			return false, nil
+		}
+
+	}
+
+	return true, nil
+}
```

Comment:
As you suggested, this could maybe be more generic and be moved next to the filter declaration.